cmake/build/compile_options.cmake

```diff
@@ -20,7 +20,10 @@ function(az_add_compile_options _TARGET)
       set(WARNINGS_AS_ERRORS_FLAG "-Werror")
     endif()
 
-    target_compile_options(${_TARGET} PRIVATE -Xclang -Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wdocumentation -Wdocumentation-unknown-command -fcomment-block-commands=retval -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wfloat-equal)
+    if(check_docs)
+      target_compile_options(${_TARGET} PRIVATE -Wdocumentation -Wdocumentation-unknown-command)
+    endif()
```

Comment:
how about doing this check always.

Reply:
there are bunch of errors in the freertos docs which get flagged. I put them in and grep out the azure failures if I'm checking for doc cleanliness. To check for other clang compiler features, I pull this out and just build since I would always get the freertos failures. Until they (or we) fix the freertos ones this is how I thought about doing it